src/hotspot/share/prims/universalUpcallHandler.cpp

```diff
@@ -36,6 +36,7 @@ extern struct JavaVM_ main_vm;
 
 void ProgrammableUpcallHandler::upcall_helper(JavaThread* thread, jobject rec, address buff) {
   JavaThread* THREAD = thread;
+  MACOS_AARCH64_ONLY(ThreadWXEnable wx(WXWrite, thread));
```

Comment:
This better be moved to `ProgrammableUpcallHandler::attach_thread_and_do_upcall` which calls ProgrammableUpcallHandler::upcall_helper; and which is called directly from assembly 

cpu/aarch64/universalUpcallHandler_aarch64.cpp
```
  __ movptr(rscratch1, CAST_FROM_FN_PTR(uint64_t, ProgrammableUpcallHandler::attach_thread_and_do_upcall));
  __ blr(rscratch1);
```